rfcs/text/0000-rfc-library-preparation.md

```diff
@@ -0,0 +1,428 @@
+
+
+---
+
+<img src="/rfcs/images/0000-lib_prep_rfc_fig5.png" align="left" height="460" />
+
+**Figure 5: Determining logical units from plate-based experimental designs**. A) An experiment modeled using the current metadata model which depicts six sequence files derived from three single cell suspension. Logical units are indicated (red, blue, and magenta outlines). B) An experiment modeled using the proposed metadata model which depicts six sequence files derived from three single cell library preparations. Logical units are indicated (red, blue, and magenta outlines). Arrows in graphs represent processes. 
+ 
+**NB**: An experimental design where the same library preparation is sequenced more than once is not shown here. This is either (1) not even feasible or (2) highly unlikely to occur given that there isn’t a lot of material to sequence the same cell more than once.
+
+---
+
+This proposal will require modifications to some DCP components (see "Implementation changes for DCP components" section below for more details). Importantly, with a new library preparation entity, the logical unit of data should be anchored on the library preparation, not the ultimate process used to generate a sequence file (colloquially referred to as the "assay process" in the Ingestion Service).
+
+### Metadata schema for library preparation entity
+
+#### Standard biomaterial fields 
+
+The new `library_preparation.json` schema will follow the schema structure indicated in the Metadata Schema Style Guide and will include the standard properties (fields) for biomaterial type schemas, including:
+
+```
+"describedBy":  {
+    "description": "The URL reference to the schema.",
+},
+"schema_version": {
+    "description": "The version number of the schema in major.minor.patch format.",
+},
+"schema_type": {
+    "description": "The type of the metadata schema entity.",
+},
+"provenance" : {
+    "description": "Provenance information provided by the system.",
+},
+"biomaterial_core" : {
+    "description": "Core biomaterial-level information.",
+}
+```
+ 
+Where the `biomaterial_core.json` schema contains the following properties:
+
+```
+"describedBy": {
+    "description": "The URL reference to the schema.",
+},
+"biomaterial_id":{
+    "description": "A [project-wide] unique ID for the biomaterial.",
+},
+"biomaterial_name": {
+    "description": "A short, descriptive name for the biomaterial that need not be unique.",
+},
+"biomaterial_description": {
+    "description": "A general description of the biomaterial.",
+},
+"ncbi_taxon_id" : {
```

Comment:
+1 to  Marcus' comment. I would recommend removing the taxon ID from anything down from the donor, unless there's a reason or occasion for this to be necessary.

Reply:
Technically, `ncbi_taxon_id` gets added to each biomaterial because it is in the `biomaterial_core` schema, which is inherited by all biomaterials. I proposed adding `ncbi_taxon_id` to the library preparation entity (at least for now) to be consistent with all other biomaterials, which all contain `biomaterial_core` and thus all have `ncbi_taxon_id`.

Reply:
There are legitimate, if rare, reasons to have ncbi taxon id on both donor and specimen, such as a humanized mouse with a human tumour but I agree that downwards from specimen it would seem unlikely. I know @simonjupp had plans which might make it easier to remove the field from other places, we could certainly create convenience tools to make it easier for contributors to not need to provide that info more times than needed.

Those feel like things which should go into a separate feature request though and not derail this process

Reply:
I'll add another note below the schema about considering other ways to populate this field so it's easier for contributors, but I will keep the recommendation about including `biomaterial_core` in the new LP schema.